src/passes/stringify-walker-impl.h

```diff
@@ -43,7 +43,12 @@ template<typename SubType>
 inline void StringifyWalker<SubType>::scan(SubType* self, Expression** currp) {
   Expression* curr = *currp;
   if (Properties::isControlFlowStructure(curr)) {
-    self->controlFlowQueue.push(currp);
+    Expression* copy = curr;
+    // To ensure control flow children are walked consistently during outlining,
+    // we push a copy of the control flow expression. This avoids an issue where
+    // control flow no longer points to the same expression after being
+    // outlined into a new function.
+    self->controlFlowQueue.push(copy);
```

Comment:
Since we are passing the `Expression*` by value, we are already creating a copy of it when we call `push()`, so we don't need to explicitly create a separate copy of it before hand.

It would be good to move the comment to the declaration of `controlFlowQueue` as well and more explicitly explain why we are storing `Expression*` rather than `Expression**`, as a reader familiar with the normal traversal utilities might expect.

```suggestion
    // To ensure control flow children are walked consistently during outlining,
    // we push a copy of the control flow expression. This avoids an issue where
    // control flow no longer points to the same expression after being
    // outlined into a new function.
    self->controlFlowQueue.push(curr);
```